packages/peregrine/src/hooks/useDOMQuery.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const findDOMElements = querySelector =>
+    document.querySelectorAll(querySelector) || [];
+
+/**
+ * A custom hook that will return the array of elements that
+ * match the querySelector. document.querySelectorAll is used
+ * to find all elements.
+ *
+ * @param {string} querySelector
+ * @return Array<HTMLElement> and {
+ *  setInnerHTML: (string) => void,
+ *  setInnerText: (string) => void,
+ *  setAttribute: (string, any) => void
+ * }
+ */
+export const useDOMQuery = querySelector => {
+    const [elements, setElements] = useState([]);
+    useEffect(() => {
+        setElements(findDOMElements(querySelector));
+    }, []);
```

Comment:
The array of dependencies is missing `querySelector` here, so the effect would never run again, even if `querySelector` changes.

Reply:
That is by design I feel. User should not be using the same hook for different queries. User can create different hooks for different query selectors. Let me know if that design needs to change.